CameraOverlay/commons.py

```diff
@@ -1,47 +1,71 @@
 import os
 import json
+import random
+import fnmatch
 from dotenv import load_dotenv
 
-from .topics import *
-
 load_dotenv()
 CONFIG_FILE = 'configs.json'
 ACTIVE_OVERLAY_KEY = 'activeOverlay'
+OVERLAY_FILE_PATTERN = 'overlay_*.py'
 CURRENT_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
 
 
-def set_overlay(new_overlays, dir=CURRENT_DIRECTORY):
+def get_overlays(directory=CURRENT_DIRECTORY):
+	overlays = [file for file in os.listdir(directory) if fnmatch.fnmatch(file, OVERLAY_FILE_PATTERN)]
+	return overlays
+
+
+def set_overlay(new_overlays, directory=CURRENT_DIRECTORY):
 	current_device = os.getenv('MHP_CAMERA')
 	new_overlay = new_overlays[current_device]
 
-	configs = read_configs(dir)
+	configs = read_configs(directory)
 	if 'device' in configs:
 		configs.pop('device')
 	configs[ACTIVE_OVERLAY_KEY] = new_overlay
 
-	with open(os.path.join(dir, CONFIG_FILE), 'w') as f:
+	with open(os.path.join(directory, CONFIG_FILE), 'w') as f:
 		json.dump(configs, f, indent=2, sort_keys=True)
 
 
-def read_configs(dir=CURRENT_DIRECTORY):
-	filepath = os.path.join(dir, CONFIG_FILE)
-	if not os.path.isfile(filepath):
-		return {}
-	else:
-		with open(filepath) as f:
+def read_configs(directory=CURRENT_DIRECTORY):
+	configs_file = os.path.join(directory, CONFIG_FILE)
+
+	if not os.path.isfile(configs_file):
+		create_default_configs(directory)
+
+	with open(configs_file) as f:
+		configs = json.load(f)
+
+	if ACTIVE_OVERLAY_KEY not in configs:
+		create_default_configs(directory)
+		with open(configs_file) as f:
 			configs = json.load(f)
 
-			current_device = os.getenv('MHP_CAMERA')
-			configs['device'] = current_device
-			return configs
+	current_device = os.getenv('MHP_CAMERA')
+	configs['device'] = current_device
+	configs['overlays'] = get_overlays()
+	return configs
+
+
+def create_default_configs(directory):
+	random_overlay = random.choice(get_overlays(directory))
```

Comment:
Not a bad idea for now, but later on we should decide on some default overlays for the rider! 👍

Reply:
yeah, i just thought hardcoding it might not be the best idea since we might change the filename later and then we'd also have to change the variables

Reply:
Yeah good idea. Let's just wait til we settle on naming conventions, default overlays etc.